### ISSUE TITLE
Entity reference causes DocumentFragment to lowercase attribute names

### ISSUE THREAD

**Issue by bhollis** [sparklemotion/nokogiri#961]
Using Nokogiri 1.6.0 on Ruby 2.0.0p0 (OSX)

``` ruby
require 'nokogiri'

xml = '<foo>
<svg viewBox="0 0 30 16"/>
Hey what?
</foo>'

bad_xml = '<foo>
<svg viewBox="0 0 30 16"/>
Hey &otimes; what?
</foo>'

d = Nokogiri::XML::Document.new
frag = Nokogiri::XML::DocumentFragment.new(d, xml, d)
bad_frag = Nokogiri::XML::DocumentFragment.new(d, bad_xml, d)
puts "GOOD: #{frag.to_s}"
puts "BAD: #{bad_frag.to_s}"

parse = Nokogiri::XML::DocumentFragment.parse(bad_xml)
puts "BONUS: #{parse.to_s}"
```

This prints:

```
GOOD: <foo>
<svg viewBox="0 0 30 16"/>
Hey what?
</foo>
BAD: <foo><svg viewbox="0 0 30 16"/>
Hey &#x2297; what?
</foo>
BONUS: <foo>
<svg viewBox="0 0 30 16"/>
Hey  what?
</foo>
```

What's happening here is that adding the `&otimes;` reference causes the document fragment to lowercase the `viewBox` attribute!

As a bonus, if I use the normal `DocumentFragment.parse` method, I just don't get any entity at all.

It's important to use an "uncommon" entity reference - stuff like `&amp;` or `&rarr;` doesn't trigger this bug.

**Comment by stadelmanma** [comment]
Was this ever determined to be a bug or expected behavior? Since needing to add `.gsub(/&(?!amp;)/, '&amp;')` in odd places doesn't seem like a good fix to  me.

**Comment by flavorjones** [comment]
This is fixed on `main` by 38c2f16, see #1158 for a related bug report and additional context. This will be fixed in the next release of Nokogiri, v1.13.0 (sometime in the next few days).
